### PR TITLE
Standard Library as JV Files

### LINKED ISSUE #397: [FEATURE] Standard Library
## User Story
1. As a {jayvee programmer}
2. I want {to reuse some valuetypes that are most commonly used}
3. So that {I don't need to redefine them from scratch}

## User Acceptance Criteria
- [x] There is a concept for a standard library in Jayvee
- [x] At least one reusable component as proof of concept
- [x] VSCode allows navigation to the definition of the components
- [x] Auto-completion for components in standard library
- [x] Standard library is structured in multiple `jv` files

## Definitions of Done
- [x] A PR has been opened and accepted
- [x] All user acceptance criteria are met
- [x] All tests are passing

### PR DESCRIPTION
**Pull request by georg-schwarz**
Added a page in the docs that also describes the contents of this PR. 
Closes #397.

Tldr:
- we can write `.jv` files into a specific directory, 
- `npm run generate` generates a TypeScript file next to the ast generation by langium, containing all jayvee files as strings
- We use this file to configure langium and vscode so that the files are used for validation, execution, and navigation in VSCode

image
image

### CODE REVIEW

File: tools/scripts/language-server/generate-stdlib.mjs [hunk 1]
```diff
@@ -0,0 +1,73 @@
+// SPDX-FileCopyrightText: 2023 Friedrich-Alexander-Universitat Erlangen-Nurnberg
+//
+// SPDX-License-Identifier: AGPL-3.0-only
+//@ts-check
```
Comment (rhazn) [inline tools/scripts/language-server/generate-stdlib.mjs:4]: What is this?

**Review by rhazn — APPROVED** [review]
Cool!

File: tools/scripts/language-server/generate-stdlib.mjs (hunk 1 above)
Comment (georg-schwarz) [reply tools/scripts/language-server/generate-stdlib.mjs:4]: [Enables the typescript checking on js files :) 
](https://www.typescriptlang.org/docs/handbook/intro-to-js-ts.html)
